src/main/scala/rocket/Consts.scala

```diff
@@ -2,80 +2,81 @@
-  def N = BitPat("b0")
-  def Y = BitPat("b1")
+  def X = BitPat.dontCare(1)
+  def N = BitPat.N()
+  def Y = BitPat.Y()
 
   val SZ_DW = 1
   def DW_X  = X
-  def DW_32 = Bool(false)
-  def DW_64 = Bool(true)
+  def DW_32 = false.B
+  def DW_64 = true.B
   def DW_XPR = DW_64
 }
 
 trait MemoryOpConstants {
   val NUM_XA_OPS = 9
   val M_SZ      = 5
-  def M_X       = BitPat("b?????");
-  def M_XRD     = UInt("b00000"); // int load
-  def M_XWR     = UInt("b00001"); // int store
-  def M_PFR     = UInt("b00010"); // prefetch with intent to read
-  def M_PFW     = UInt("b00011"); // prefetch with intent to write
-  def M_XA_SWAP = UInt("b00100");
-  def M_FLUSH_ALL = UInt("b00101")  // flush all lines
-  def M_XLR     = UInt("b00110");
-  def M_XSC     = UInt("b00111");
-  def M_XA_ADD  = UInt("b01000");
-  def M_XA_XOR  = UInt("b01001");
-  def M_XA_OR   = UInt("b01010");
-  def M_XA_AND  = UInt("b01011");
-  def M_XA_MIN  = UInt("b01100");
-  def M_XA_MAX  = UInt("b01101");
-  def M_XA_MINU = UInt("b01110");
-  def M_XA_MAXU = UInt("b01111");
-  def M_FLUSH   = UInt("b10000") // write back dirty data and cede R/W permissions
-  def M_PWR     = UInt("b10001") // partial (masked) store
-  def M_PRODUCE = UInt("b10010") // write back dirty data and cede W permissions
-  def M_CLEAN   = UInt("b10011") // write back dirty data and retain R/W permissions
-  def M_SFENCE  = UInt("b10100") // SFENCE.VMA
-  def M_HFENCEV = UInt("b10101") // HFENCE.VVMA
-  def M_HFENCEG = UInt("b10110") // HFENCE.GVMA
-  def M_WOK     = UInt("b10111") // check write permissions but don't perform a write
-  def M_HLVX    = UInt("b10000") // HLVX instruction
+  def M_X       = BitPat.dontCare(5)
+  def M_XRD     = "b00000".U(5.W) // int load
+  def M_XWR     = "b00001".U(5.W) // int store
+  def M_PFR     = "b00010".U(5.W) // prefetch with intent to read
+  def M_PFW     = "b00011".U(5.W) // prefetch with intent to write
+  def M_XA_SWAP = "b00100".U(5.W)
+  def M_FLUSH_ALL = "b00101".U(5.W)  // flush all lines
+  def M_XLR     = "b00110".U(5.W)
+  def M_XSC     = "b00111".U(5.W)
+  def M_XA_ADD  = "b01000".U(5.W)
+  def M_XA_XOR  = "b01001".U(5.W)
+  def M_XA_OR   = "b01010".U(5.W)
+  def M_XA_AND  = "b01011".U(5.W)
+  def M_XA_MIN  = "b01100".U(5.W)
+  def M_XA_MAX  = "b01101".U(5.W)
+  def M_XA_MINU = "b01110".U(5.W)
+  def M_XA_MAXU = "b01111".U(5.W)
+  def M_FLUSH   = "b10000".U(5.W) // write back dirty data and cede R/W permissions
+  def M_PWR     = "b10001".U(5.W) // partial (masked) store
+  def M_PRODUCE = "b10010".U(5.W) // write back dirty data and cede W permissions
+  def M_CLEAN   = "b10011".U(5.W) // write back dirty data and retain R/W permissions
+  def M_SFENCE  = "b10100".U(5.W) // SFENCE.VMA
+  def M_HFENCEV = "b10101".U(5.W) // HFENCE.VVMA
+  def M_HFENCEG = "b10110".U(5.W) // HFENCE.GVMA
+  def M_WOK     = "b10111".U(5.W) // check write permissions but don't perform a write
+  def M_HLVX    = "b10000".U(5.W) // HLVX instruction
 
```

Comment:
All widths in this file should similarly no longer be necessary right? Can we undo the change just to minimize the diff?

Reply:
Yes, but it seems that chipsalliance/chisel3#2387 didn't catch the 3.5.1...